paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -367,11 +364,14 @@ internal class PaymentSheetViewModel @Inject internal constructor(
                 eventReporter.onPaymentFailure(selection.value)
 
                 runCatching {
-                    stripeIntentValidator.requireValid(stripeIntentResult.intent)
+                    stripeIntentValidator.requireValid(stripeIntent)
                 }.fold(
                     onSuccess = {
                         resetViewState(
-                            stripeIntentResult.failureMessage
+                            when (paymentResult) {
+                                is PaymentResult.Failed -> paymentResult.throwable.message
+                                else -> "" // indicates canceled payment
```

Comment:
I think this should be `null` [instead](https://github.com/stripe/stripe-android/blob/3c2adea9c9739e0d77125e0813dea596d8dc6905/payments-core/src/main/java/com/stripe/android/payments/PaymentFlowFailureMessageFactory.kt#L32)

Reply:
fixed in [1703bef](https://github.com/stripe/stripe-android/pull/4159/commits/1703bef6d382a02819bebae65cd66be7e6658e81)